### PR TITLE
fix: Bump Go and Kind Versions

### PR DESCRIPTION
**Pull request by MeenuyD**
Kind version change to test version v1.26 to v.1.28
Go version is bumped to v1.21
#961

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -11,7 +11,7 @@ replace (
 require (
 	github.com/google/go-cmp v0.5.9
 	github.com/projectcontour/contour v1.25.0
-	go.uber.org/zap v1.25.0
+	go.uber.org/zap v1.26.0
```
Comment (dprotaso) [inline go.mod:14]: you'll have to roll this change back to v1.25.0

or run `./hack/update-deps` script

**Comment by dprotaso** [comment]
@MeenuyD can you update this PR please?

File: go.sum [hunk 2]
```diff
@@ -344,7 +344,7 @@ go.uber.org/multierr v1.10.0 h1:S0h4aNzvfcFsC3dRF1jLoaov7oRaKqRGC/pUEJ2yvPQ=
 go.uber.org/multierr v1.10.0/go.mod h1:20+QtiLqy0Nd6FdQB9TLXag12DsQkrbs3htMFfDN80Y=
 go.uber.org/zap v1.10.0/go.mod h1:vwi/ZaCAaUcBkycHslxD9B2zi4UTXhF60s6SWpuDF0Q=
 go.uber.org/zap v1.25.0 h1:4Hvk6GtkucQ790dqmj7l1eEnRdKm3k3ZUrUMS2d5+5c=
-go.uber.org/zap v1.25.0/go.mod h1:JIAUzQIH94IC4fOJQm7gMmBJP5k7wQfdcnYdPoEXJYk=
+go.uber.org/zap v1.25.0 /go.mod h1:JIAUzQIH94IC4fOJQm7gMmBJP5k7wQfdcnYdPoEXJYk=
```
Comment (dprotaso) [inline go.sum:347]: This edit seems unnecessary

**Review by dprotaso — CHANGES_REQUESTED** [review]

File: go.sum [hunk 3]
```diff
@@ -343,8 +343,8 @@ go.uber.org/multierr v1.1.0/go.mod h1:wR5kodmAFQ0UK8QlbwjlSNy0Z68gJhDJUG5sjR94q/
 go.uber.org/multierr v1.10.0 h1:S0h4aNzvfcFsC3dRF1jLoaov7oRaKqRGC/pUEJ2yvPQ=
 go.uber.org/multierr v1.10.0/go.mod h1:20+QtiLqy0Nd6FdQB9TLXag12DsQkrbs3htMFfDN80Y=
 go.uber.org/zap v1.10.0/go.mod h1:vwi/ZaCAaUcBkycHslxD9B2zi4UTXhF60s6SWpuDF0Q=
-go.uber.org/zap v1.25.0 h1:4Hvk6GtkucQ790dqmj7l1eEnRdKm3k3ZUrUMS2d5+5c=
-go.uber.org/zap v1.25.0/go.mod h1:JIAUzQIH94IC4fOJQm7gMmBJP5k7wQfdcnYdPoEXJYk=
+go.uber.org/zap v1.26.0 h1:4Hvk6GtkucQ790dqmj7l1eEnRdKm3k3ZUrUMS2d5+5c=
```
Comment (dprotaso) [inline go.sum:346]: Leave the go sum file untouched - so revert the changes here

**Review by dprotaso — CHANGES_REQUESTED** [review]

File: go.sum (hunk 3 above)
Comment (MeenuyD) [reply go.sum:346]: I am a little bit confused can you please tell me should the go.uber.org/zap version is 1.25.0 or 1.26.0? It will be very helpful for me.

Comment (dprotaso) [reply go.sum:346]: This file is unrelated to bumping the k8s version. So I would undo the changes here so it doesn't show up in this PR

**Comment by MeenuyD** [comment]
can you please review #977
